fortinet_fortigate/icon_fortinet_fortigate/actions/delete_address_object/action.py

```diff
@@ -34,11 +35,22 @@ def run(self, params={}):
                 )
             pass
 
-        params_payload = {"mkey": str(host)}
+        is_ipv6 = self.connection.get_address_object(original_host)["name"] == "address6"
 
-        endpoint = f"https://{self.connection.host}/api/v2/cmdb/firewall/address"
-
-        response = self.connection.session.delete(endpoint, params=params_payload, verify=self.connection.ssl_verify)
+        if is_ipv6:
+            params_payload = {"mkey": original_host}
+            response = self.connection.session.delete(
```

Comment:
use 'response.raise_for_status()` and return reasonable error messages if an exception is raised.

Reply:
As far as I can see, there is a try/catch below, where .json() is called. Isn't that good enough? As we have only added a condition here for IPv6, I guess it should be fine?

Reply:
@tfranckiewicz-r7 the `try` block below will only catch an exception if the response is not JSON - it will not handle other issues that may arise (timeouts, bad request, server error, etc.)

Reply:
I changed this line to use method: call_api where we use try/except.

Reply:
Perfect 👍